qiskit/circuit/library/arithmetic/adders/adder.py

```diff
@@ -208,3 +226,9 @@ def num_state_qubits(self) -> int:
             The number of state qubits.
         """
         return self._num_state_qubits
+
+    def _define(self):
+        """Populates self.definition with a decomposition of this gate."""
+        from qiskit.synthesis.arithmetic import adder_ripple_c04
+
+        self.definition = adder_ripple_c04(self.num_state_qubits, kind="full")
```

Comment:
It would be nice to add a comment here that in the special case of a "full" adder, this method doesn't use ancillas -- otherwise it seems a bit surprising that this is the only one not using the QFT-based decomp

Reply:
Added this comment.